elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/FieldHelper.java

```diff
@@ -101,6 +101,76 @@ static boolean isCalendar(EntityIndexBinding indexBinding, String fieldName) {
 		return Calendar.class.isAssignableFrom( propertyClass );
 	}
 
+	static boolean isInstant(EntityIndexBinding indexBinding, String fieldName) {
+		Class<?> propertyClass = getPropertyClass( indexBinding, fieldName );
```

Comment:
Yes, it's always difficult to decide when to change an existing pattern. I think the fact that you added a lot more instances and that we can now rely on an enum for the types is a good reason to change that. That being said, maybe you should wait for https://github.com/hibernate/hibernate-search/pull/1151/commits to be merged to change that?